src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs

```diff
@@ -129,7 +135,10 @@ protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textViewOpt
         }
 
         // Otherwise, we need to go produce all tags.
-        var options = _globalOptions.GetHighlightingOptions(document.Project.Language);
+        var options = _globalOptions.GetHighlightingOptions(document.Project.Language) with
+        {
+            FrozenPartialSemantics = context.FrozenPartialSemantics,
+        };
         return ProduceTagsAsync(context, caretPosition, document, options, cancellationToken);
```

Comment:
📝 The same applies here regarding `context` and `options`

Reply:
done.